lib/layout.js

```diff
@@ -190,17 +198,27 @@ const pickZone = (opts, frameBounds, targetBounds, size) => {
 
   /* If the preferred side is part of the available zones, use that otherwise
   pick the largest available zone. If there are no available zones, pick the
-  largest zone. TODO: logic that executes picking based on largest option. */
+  largest zone. */
 
   if (opts.preferPlace) {
     const preferenceType = getPreferenceType(opts.preferPlace)
     if (!preferenceType) throw createPreferenceError(opts.preferPlace)
+    
+    // Try to fit first in zone where the pop up fit completely
     const preferredAvailZones = availZones.filter((zone) => (
       zone[preferenceType] === opts.preferPlace
     ))
     if (preferredAvailZones.length) return preferredAvailZones[0]
+    
+    // If there are not areas where the pop up fit completely, it uses the prefered ones
+    // in order from the one the fit better
+    const preferredZones = zones.filter((zone) => (
+      zone[preferenceType] === opts.preferPlace
+    ))
+    if (preferredZones.length) return preferredZones[0]    
```

Comment:
Hm, this seems like a regression, @jasonkuhrt, @ChrisCinelli?

This will pick from `preferredZones` ignoring whether or not there are other available zones (which is a case handled by the return statement after this). To match what the comment says, it seems like this check should be
```diff
- if (preferredZones.length) return preferredZones[0]  
+ if (!availZones.length && preferredZones.length) return preferredZones[0]  
```

Here is a gif of the behavior I'm seeing (with `preferPlace` as `below`), which is rectified by the above diff:
![falling-off-issue](https://cloud.githubusercontent.com/assets/2320890/13201352/57c0ab6e-d822-11e5-90b9-84ae80e744ed.gif)